src/compiler/checker.ts

```diff
@@ -13443,7 +13443,7 @@ namespace ts {
         }
 
         function getTypeOfFirstParameterOfSignature(signature: Signature) {
-            return getTypeAtPosition(signature, 0);
+            return getTypeWithFacts(getTypeAtPosition(signature, 0), TypeFacts.NEUndefined);

```

Comment:
this function is used in two places 
- when we get type of first parameter in the call signature of `then`
- when we get type of first parameter in the `onFulfilled` callback. 

Do we want to strip `undefined` in both places or just in the first one?